misk-jobqueue/src/main/kotlin/misk/jobqueue/QueueKey.kt

```diff
@@ -0,0 +1,9 @@
+package misk.jobqueue
+
+/**
+ * Represents an ordered list of queue names, for building a priority queue using underlying queues.
+ * Allows us to workaround limitations with Guice's handling of generics.
```

Comment:
what does generics have to do with it?

Reply:
Guice doesn't support binding to a generic key (ie List<QueueName>) in Kotlin without some pretty ugly annotations. Comment is probably more confusing than helpful though, so I'll just remove it.

Reply:
hmm ok. we do it all over the place in misk. do you mean just adding the `@JvmWildcard`?

Reply:
yes. it was jesse's suggestion to pull out a class. i don't feel that strongly about it, though the @JvmWildcard is pretty ugly imo.